modules/Core/includes/endpoints/RegisterEndpoint.php

```diff
@@ -0,0 +1,241 @@
+
+        $minecraft_integration = Util::getSetting($api->getDb(), 'mc_integration');
+        if ($minecraft_integration) $params[] = 'uuid';
+
+        if ($api->validateParams($_POST, $params)) {
+            
+            if ($minecraft_integration) {
+                $_POST['uuid'] = str_replace('-', '', $_POST['uuid']);
+                if (strlen($_POST['uuid']) > 32) $api->throwError(9, $api->getLanguage()->get('api', 'invalid_uuid'));
+            }
+
+            if (strlen($_POST['username']) > 20) $api->throwError(8, $api->getLanguage()->get('api', 'invalid_username'));
+            if (strlen($_POST['email']) > 64) $api->throwError(7, $api->getLanguage()->get('api', 'invalid_email_address'));
+
+            // Validate email
+            if (!filter_var($_POST['email'], FILTER_VALIDATE_EMAIL)) $api->throwError(7, $api->getLanguage()->get('api', 'invalid_email_address'));
+
+            // Ensure user doesn't already exist
+            $username = $api->getDb()->get('users', array('username', '=', Output::getClean($_POST['username'])));
+            if (count($username->results())) $api->throwError(11, $api->getLanguage()->get('api', 'username_already_exists'));
+
+            if ($minecraft_integration) {
+                $uuid = $api->getDb()->get('users', array('uuid', '=', Output::getClean($_POST['uuid'])));
+                if (count($uuid->results())) $api->throwError(12, $api->getLanguage()->get('api', 'uuid_already_exists'));
+            }
+
+            $email = $api->getDb()->get('users', array('email', '=', Output::getClean($_POST['email'])));
+            if (count($email->results())) $api->throwError(10, $api->getLanguage()->get('api', 'email_already_exists'));
+
+            $uuid = ($minecraft_integration) ? Output::getClean($_POST['uuid']) : 'none';
+
+            // Registration email enabled?
+            if (Util::getSetting($api->getDb(), 'email_verification', true)) {
+                // Send email
+                $this->sendRegistrationEmail($api, $_POST['username'], $uuid, $_POST['email']);
+            } else {
+                // Register user + send link
+                $code = $this->createUser($api, $_POST['username'], $uuid, $_POST['email']);
+                $api->returnArray(array('message' => $api->getLanguage()->get('api', 'finish_registration_link'), 'link' => rtrim(Util::getSelfURL(), '/') . URL::build('/complete_signup/', 'c=' . $code['code'])));
+            }
+        }
+    }
+
+    /**
+     * Sends verification email upon new registration
+     * 
+     * For internal API use only
+     * 
+     * @see Nameless2API::register()
+     * 
+     * @param string $username The username of the new user to create
+     * @param string $uuid (optional) The Minecraft UUID of the new user
+     * @param string $email The email of the new user
+     * 
+     * @return string JSON Array
+     */
+    private function sendRegistrationEmail(Nameless2API $api, $username, $uuid, $email) {
```

Comment:
Another type

Reply:
As per our conversation on Discord, leaving this.